src/components/ProposalStatusForm.tsx

```diff
@@ -14,6 +15,8 @@ export default function ProposalStatusForm({
   proposalDetails,
   session,
 }: ProposalStatusFormProps) {
+  const [feedbackGiven, setFeedbackGiven] = useState(false)
```

Comment:
the state needs to be "proposal-specific". it is counterproductive if, after giving one ACCEPT, all other ACCEPT forms are blocked. using the, e.g., useSessionStorage hook, this could be simple: just use the proposalId as the "key" (first arg), and store something like the feedback type given (e.g., REJECT). you can hide the form and show a message when either the value in session storage is set (after submission but before the flow has finished/page reloaded) or a previous feedback has been fetched from the server (after the flow has run). TENTATIVE would be the only special case, where forms have to be displayed.